specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01/redis.json

```diff
@@ -1850,85 +1698,583 @@
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
         }
       },
-      "required": [
-        "name",
-        "family",
-        "capacity"
-      ],
-      "description": "SKU parameters supplied to the create Redis operation."
-    },
-    "RedisCommonProperties": {
-      "properties": {
-        "redisConfiguration": {
-          "type": "object",
-          "properties": {
-            "rdb-backup-enabled": {
-              "type": "string",
-              "description": "Specifies whether the rdb backup is enabled"
-            },
-            "rdb-backup-frequency": {
-              "type": "string",
-              "description": "Specifies the frequency for creating rdb backup in minutes. Valid values: (15, 30, 60, 360, 720, 1440)"
-            },
-            "rdb-backup-max-snapshot-count": {
-              "type": "string",
-              "description": "Specifies the maximum number of snapshots for rdb backup"
-            },
-            "rdb-storage-connection-string": {
-              "type": "string",
-              "description": "The storage account connection string for storing rdb file"
-            },
-            "aof-backup-enabled": {
-              "type": "string",
-              "description": "Specifies whether the aof backup is enabled"
-            },
-            "aof-storage-connection-string-0": {
-              "type": "string",
-              "description": "First storage account connection string"
-            },
-            "aof-storage-connection-string-1": {
-              "type": "string",
-              "description": "Second storage account connection string"
-            },
-            "maxfragmentationmemory-reserved": {
-              "type": "string",
-              "description": "Value in megabytes reserved for fragmentation per shard"
+      "delete": {
+        "tags": [
+          "Redis",
+          "AAD"
+        ],
+        "operationId": "AccessPolicy_Delete",
+        "x-ms-examples": {
+          "RedisCacheAccessPolicyDelete": {
+            "$ref": "./examples/RedisCacheAccessPolicyDelete.json"
+          }
+        },
+        "description": "Deletes the access policy from a redis cache",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "cacheName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Redis cache.",
+            "pattern": "^([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]|[a-zA-Z0-9])$"
```

Comment:
Why not refer the global parameter here? https://github.com/Azure/azure-rest-api-specs/blob/813022a095cd9cfd919795167689053a286890d4/specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01/redis.json#L3356

Reply:
@MaryGao if I add a pattern property to the global CacheNameParameter then it's considered as a breaking change for all the operations that didn't have a pattern defined in the past.
If I remove the pattern property from the CacheNameParameter and try to use it for all cases, then the latest operations that I've added fail validation, stating that a pattern property must be added.

Thus I moved forward with defining cacheName again for the new operations, with the pattern property.

Reply:
As per discussion added pattern to global cacheNameParameter